Controllers/FrontController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Controllers;
+
+class FrontController
+{
+    protected $controller;
+    protected $action, $params;
+    static $instance;
+
+    public static function getInstance()
+    {
+        if (!(self::$instance instanceof self)) {
+            self::$instance = new self();
+        }
+        return self::$instance;
+    }
+
+    private function __construct()
+    {
+        $request = $_SERVER['REQUEST_URI'];
+        $splits = explode('/', trim($request, '/'));
+        $this->controller = !empty($splits[0]) ?
+            $this->getFullControllerName(ucfirst($splits[0]) . 'Controller')
+            : $this->getFullControllerName('ViewController');

```

Comment:
"ViewController" sounds not clear...
I'd rather name default controller as **IndexController**